.travis.yml

```diff
@@ -0,0 +1,31 @@
+services:
+- docker
+
+before-install:
+  - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+  - docker login --username=$HEROKU_LOGIN --password=$HEROKU_API_KEY registry.heroku.com
+
+before-script:
+  - docker build --target test --tag my-test-image .
+  - docker build --tag $DOCKER_USERNAME/todo-app:latest --target production .
+
+script:
+  - docker run my-test-image tests/test_viewmodel.py
+  - docker run my-test-image tests/test_integration.py
+  - >
+     docker run
+     -e MONGO_DB_CONNECTION=$MONGO_DB_CONNECTION
+     -e MONGO_DB_DATABASE=$MONGO_DB_DATABASE
+     my-test-image tests/test_e2e.py
+
+after-script:
+  - docker push $DOCKER_USERNAME/todo-app:latest
+
+deploy:
+  provider: script
+  script: 
+        docker tag 180291/todo-app registry.heroku.com/my-todo-app-1/web;
```

Comment:
Is 180291 your Docker Hub username? Not a big deal to hardcode it, but seems like a shame when you use `$DOCKER_USERNAME` everywhere else!